### PR TITLE
Add from_axes for Box entities

### PR DESCRIPTION
**Pull request by yifan-flex**

### CODE REVIEW

File: flow360/component/simulation/primitives.py [hunk 1]
```diff
@@ -122,7 +138,85 @@ class Box(_VolumeEntityBase):
...
+            name=name,
+            center=center,
+            size=size,
+            axis_of_rotation=list(axis),
+            angle_of_rotation=angle * u.rad,
+        )
+
+    # pylint: disable=no-self-argument
+    @classmethod
+    def _rotation_matrix_from_axis_and_angle(cls, axis, angle):
```
Comment (benflexcompute) [inline flow360/component/simulation/primitives.py:186]: Can we move this out of box class as a standalone function?

File: flow360/component/simulation/primitives.py [hunk 2]
```diff
@@ -122,7 +138,85 @@ class Box(_VolumeEntityBase):
...
+        # Compute the skew-symmetric cross-product matrix of axis
+        cross_n = np.array([[0, -n_z, n_y], [n_z, 0, -n_x], [-n_y, n_x, 0]])
+
+        # Compute the rotation matrix
+        rotation_matrix = np.eye(3) + sin_theta * cross_n + one_minus_cos * np.dot(cross_n, cross_n)
+
+        return rotation_matrix
+
+    @pd.model_validator(mode="after")
+    def _convert_axis_and_angle_to_coordinate_axes(cls, obj):
```
Comment (benflexcompute) [inline flow360/component/simulation/primitives.py:203]: This is an instance method.

File: flow360/component/simulation/primitives.py [hunk 3]
```diff
@@ -122,7 +138,85 @@ class Box(_VolumeEntityBase):
...
+        # Ensure the axis is a numpy array
+        if not obj.private_attribute_input_cache.axes:
+            axis = np.asarray(obj.axis_of_rotation, dtype=float)
```
Comment (benflexcompute) [inline flow360/component/simulation/primitives.py:206]: Try go further with np.float64

File: tests/simulation/framework/test_entities_v2.py [hunk 4]
```diff
@@ -858,3 +859,67 @@ def test_corner_cases_for_entity_registry_thoroughness(my_cylinder1, my_volume_m
...
+            )
+        )
+    )
+
+
+def test_box_creation():
+    box1 = Box(
+        name="box1",
+        center=[0, 0, 0] * u.m,
+        size=[1, 1, 1] * u.m,
```
Comment (benflexcompute) [inline tests/simulation/framework/test_entities_v2.py:886]: Interesting. I thought list is not supported.

Comment (benflexcompute) [reply tests/simulation/framework/test_entities_v2.py:886]: Can we change and use tuple to be consistent?

Comment (benflexcompute) [reply tests/simulation/framework/test_entities_v2.py:886]: Can we add a 0 rotation test, x to y and y to x, 90 degree test? You can use update() to update only the part you need to change for less coding.

File: flow360/component/simulation/primitives.py [hunk 5]
```diff
@@ -107,8 +113,18 @@ class GenericSurface(_SurfaceEntityBase):
     )
 
 
+class BoxCache(Flow360BaseModel):
```
[the remark this answers is not in the document]
Comment (yifan-flex) [reply flow360/component/simulation/primitives.py:116]: Ok

File: flow360/component/simulation/primitives.py [hunk 6]
```diff
@@ -107,8 +113,18 @@ class GenericSurface(_SurfaceEntityBase):
...
+    """BoxCache"""
+
+    axes: Optional[Tuple[Axis, Axis]] = pd.Field(None)
```
[the remark this answers is not in the document]
Comment (yifan-flex) [reply flow360/component/simulation/primitives.py:119]: Ok so we need a new data type

File: flow360/component/simulation/primitives.py [hunk 7]
```diff
@@ -122,7 +138,85 @@ class Box(_VolumeEntityBase):
...
     size: LengthType.Point = pd.Field()
-    axes: Tuple[Axis, Axis] = pd.Field()
+    axis_of_rotation: Axis = pd.Field()
+    angle_of_rotation: AngleType = pd.Field()
+    private_attribute_input_cache: BoxCache = pd.Field(BoxCache(), frozen=True)
+
+    # pylint: disable=no-self-argument, missing-function-docstring
+    @MultiConstructorBaseModel.model_constructor
+    @pd.validate_call
+    def from_axes(
```
[the remark this answers is not in the document]
Comment (yifan-flex) [reply flow360/component/simulation/primitives.py:148]: Ok, will rename to principal axes

File: flow360/component/simulation/primitives.py [hunk 8]
```diff
@@ -122,7 +138,85 @@ class Box(_VolumeEntityBase):
...
+        cls, name: str, center: LengthType.Point, size: LengthType.Point, axes: Tuple[Axis, Axis]
+    ):
+        # validate
+        x_axis, y_axis = np.array(axes[0]), np.array(axes[1])
+        x_axis /= np.linalg.norm(x_axis)
```
[the remark this answers is not in the document]
Comment (yifan-flex) [reply flow360/component/simulation/primitives.py:153]: Yeah.. So when the case is load back the user won't see the same input. Maybe it's ok

File: flow360/component/simulation/primitives.py (hunk 1 above)
Comment (yifan-flex) [reply flow360/component/simulation/primitives.py:186]: Where should this function be?

**Review by benflexcompute — APPROVED** [review]

**Pull request merged**